Lib/fontTools/ttLib/tables/otBase.py

```diff
@@ -70,6 +77,12 @@ def compile(self, font):
 			try:
 				writer = OTTableWriter(tableTag=self.tableTag)
 				self.table.compile(writer, font)
+				if have_uharfbuzz and self.tableTag in ("GSUB", "GPOS"):
+					try:
+						return writer.getAllDataUsingHarfbuzz()
+					except (ValueError, MemoryError, hb.RepackerError) as e:
+						log.error("hb.repack failed, reverting to pure-python serializer; the error message was: %s", e)
+						return writer.getAllData(remove_duplicate=False)
```

Comment:
this except branch is currently never hit by the test suite. I know it's difficult to make the harfbuzz repacker fail, but I wonder if you can come up with a test that produces such a situation in order to trigger this and confirm that calling `writer.getAllData(remove_duplicate=False)` actually works.

Reply:
yeah, I actually tested this except branch locally by changing uharfbuzz code, but I don't have a real testcase for this. Let me try and see if I can find or make a testcase

Reply:
I tried building GSans with this branch and hb.repack() is always failing.

Reply:
> I tried building GSans with this branch and hb.repack() is always failing.

could you provide a testcase? I can send it to Garret for debugging. This is also a good testcase for except branch.

Reply:
I don’t know how to extract a test case from this, it is being built from UFO source. I get messages like this:
```
ERROR:fontTools.ttLib.tables.otBase:hb.repack failed, reverting to pure-python serializer; the error message was:
INFO:fontTools.ttLib.tables.otBase:Attempting to fix OTLOffsetOverflowError ('GPOS', 'LookupIndex:', 15, 'SubTableIndex:', None, 'ItemName:', None, 'ItemIndex:', None)
```

Reply:
It looks like harfbuzz repacker failed to find a solution so fonttools falls back to the except branch as expected.
This "OTLOffsetOverflowError" is raised during writer.getAllData().

Reply:
Yeah, the "attempting to fix offset overflow" is not an "ERROR", in fact it's marked as "INFO" in the logger output. This has been a source of confusions for ages..

Reply:
@qxliu76 are you going to add a test case that covers the except branch?

Reply:
yes, Garret said he should be able to generate a graph that fails to repack. I will likely get that next Monday and then add a testcase

Reply:
My impression is that the HarfBuzz repacker would be useful specifically in offset overflow cases where FontTools is currently too slow, so I’m just curious why it is failing here.

Reply:
This is just my guess: If subtables larger than 65kb exist, it's not possible for any offsets to point past them. In these cases the subtable can usually be split into two smaller subtables to allow for more flexibility in the ordering. Currently the harfbuzz repacker does not support table splitting. In the future we would like to explore either adding table splitting into harfbuzz repacker or finding a way to utilize fonttools table splitting before sending the graph to harfbuzz.

Reply:
@anthrotype I just added a testcase that fails to repack because harfbuzz does not support table splitting. Since fonttoolsI is slow, I noticed that runtime for subset_test.py has bumped from 0.95s to 35s.

Reply:
@anthrotype Besides runtime bump, the test file is pretty big as well, 14 MB. I wonder if we might need to find a different way to test this. Another way to setup the test would be to use something similar to the mock test, substitute the uharfbuzz object with a mock that always returns failure when the repack method is called. I'm not familiar with mocking stuff in python, not sure if you would have any suggestions.

Reply:
yeah, it's too big. It's better to mock it.
Take a look at https://docs.python.org/3/library/unittest.mock.html

or if you want, I can add the mocked test after merging this PR

Reply:
Thank you! I just removed this testcase.